resources/views/faults/list.blade.php

```diff
@@ -0,0 +1,98 @@
+            }
+            var translations = {
+                UNSEEN: "@lang('faults.unseen')",
+                SEEN: "@lang('faults.seen')",
+                DONE: "@lang('faults.done')",
+                WONT_FIX: "@lang('faults.wont_fix')"
+            };
+            var colors = {
+                UNSEEN: "text-secondary",
+                SEEN: "text-info",
+                DONE: "text-success",
+                WONT_FIX: "text-danger"
+            };
+            return '<div class="font-italic ' + colors[cell] + ' text-right">' + translations[cell] + '</div>';
+        }
+
+        var button = function (cell, formatterParams) {
+            if (formatterParams['status'] === "DONE") {
+                var style = "btn-success";
+                var text = "@lang('faults.done')";
+            } else if (formatterParams['status'] === "WONT_FIX") {
+                var style = "btn-danger";
+                var text = "@lang('faults.wont_fix')";
+            } else if (formatterParams['status'] === "UNSEEN") {
+                var style = "btn-danger";
+                var text = "@lang('faults.reopen')";
+            }
+            return $("<button type=\"button\" class=\"btn btn-sm " + style + " float-left\">" + text + "</button>").click(function () {
+                var data = cell.getRow().getData();
+                var returned;
+                confirm('@lang('faults.confirm')', '@lang('faults.confirm_message')' + text, '@lang('faults.cancel')', '@lang('faults.confirm')', function() {
+                    $.post("{{ route('faults.update') }}", {id: data.id, status: formatterParams['status']}, function(data){
+                        if (data !== "true"){
+                            alert('@lang('faults.autherror')');
+                        } else {
+                            window.location.reload(true);
+                        }
+                    });
+                });
+            })[0];
+        };
+
+        var button_formatter = function (cell, formatterParams, onRendered) {
+            var status = cell.getRow().getData()["status"];
+            if (formatterParams["status"] === "UNSEEN") {
+                if (status === "DONE" || status === "WONT_FIX") {
+                    return button(cell, formatterParams);
+                } else {
+                    return "";
+                }
+            }
+            if (status === "UNSEEN" && formatterParams["status"] === "WONT_FIX") {
+                onRendered(function () {
+                    $.post("{{ route('faults.update') }}", {id: cell.getValue(), status: "SEEN"}, );
+                });
+                return button(cell, formatterParams);
+            } else if (status === "SEEN" || status === "UNSEEN") {
+                return button(cell, formatterParams);
+            } else if (formatterParams["status"] === "WONT_FIX") {
+                return status_translate(status);
+            }
+            return "";
+        }
```

Comment:
To be honest this looks quite complicated. Can't we simplify this?

Reply:
I've changed that (now it's also a `switch`), hopefully it is more readable now.